BaseApiBundle/Repository/ApiClientRepository.php

```diff
@@ -20,4 +20,134 @@ public function findOneByKeyAndSecret($key, $secret)
     {
         return $this->findOneBy(array('key' => $key, 'secret' => $secret));
     }
+
+    /**
+     * @param array|null  $descriptionEntity
+     * @param array|null  $columns
+     * @param string|null $search
+     * @param array|null  $order
+     * @param int|null    $skip
+     * @param int|null    $limit
+     *
+     * @return array
+     */
+    public function findForPaginateAndSearch($descriptionEntity = null, $columns = null, $search = null, $order = null, $skip = null, $limit = null)
+    {
+        $qb = $this->createQueryWithSearchAndOrderFilter($descriptionEntity, $columns, $search, $order);
+
+        if (null !== $skip && $skip > 0) {
+            $qb->skip($skip);
+        }
+
+        if (null !== $limit) {
+            $qb->limit($limit);
+        }
+
+        return $qb->getQuery()->execute();
+    }
+
+    /**
+     * @return int
+     */
+    public function count()
+    {
+        $qb = $this->createQueryBuilder();
+
+        return $qb->count()->getQuery()->execute();
+    }
+
+    /**
+     * @param array|null   $columns
+     * @param array|null   $descriptionEntity
+     * @param array|null   $search
+     *
+     * @return int
+     */
+    public function countFilterSearch($descriptionEntity = null, $columns = null, $search = null)
```

Comment:
I think there is also a problem with the naming here, some might think the method is counting the quantity of filters.